### PR TITLE
fix spaces and dots in fragment id of markdown links

### PR DESCRIPTION
**Pull request by eshellman**
## Markdown doesn't allow spaces in attempted links

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -108,7 +108,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
     * [Ionic](#ionic)
     * [jQuery](#jquery)
     * [meteor](#meteor)
-    * [Next.js](#next.js)
+    * [Next.js](#nextjs)
```
Comment (davorpa) [inline books/free-programming-books-langs.md:111]: I think you fail with the right branch committing 0b5884874b69a0a23f0f5bccacf6775cce0d3e0b lol. Maybe this goes in the other PR?

Could you cherry pick this 0b5884874b69a0a23f0f5bccacf6775cce0d3e0b into #7020 (https://github.com/EbookFoundation/free-programming-books/tree/fix-caps-in-frags) and revert/hard reset this change here?

Comment (davorpa) [reply books/free-programming-books-langs.md:111]: Sorry, I missunderstand the PR attending to its title. I update it

**Comment by davorpa** [comment]
Sure there are more dots in fragments according to this regex `\(#.*\..*\)`. 

image

Do you forget submit them in commit 0b5884874b69a0a23f0f5bccacf6775cce0d3e0b or use other regex? Let me know :wink:

**Comment by eshellman** [comment]
the ones in free-courses-ko are fixed in a pending PR #7010

**Comment by davorpa** [comment]
> the ones in free-courses-ko are fixed in a pending PR #7010

I don't think so since @GH05T-HUNTER5 reverted his/her changes with f7896fab0f35ec244ebb5b7168f93e163b6745fc

Preparing a PR :wink: with the remaining over this branch

**Pull request merged**